mock-api/msw/handlers.ts

```diff
@@ -1667,6 +1699,35 @@ export const handlers = makeHandlers({
       })
     return { items: members }
   },
+  antiAffinityGroupMemberInstanceAdd({ path, query }) {
+    const project = lookup.project({ ...query })
+    const instance = lookup.instance({ ...query, instance: path.instance })
+    const antiAffinityGroup = lookup.antiAffinityGroup({
+      project: project.id,
+      antiAffinityGroup: path.antiAffinityGroup,
+    })
+    const alreadyThere = db.antiAffinityGroupMemberLists.some(
+      (i) =>
+        i.anti_affinity_group_id === antiAffinityGroup.id &&
+        i.anti_affinity_group_member.id === instance.id
+    )
+    if (alreadyThere) {
+      throw 'Instance already in anti-affinity group'
+    }
+    const newMember: Json<Api.AntiAffinityGroupMember> = {
+      type: 'instance',
+      value: {
+        id: instance.id,
+        name: instance.name,
+        run_state: instance.run_state,
+      },
+    }
+    db.antiAffinityGroupMemberLists.push({
+      anti_affinity_group_id: antiAffinityGroup.id,
+      anti_affinity_group_member: { type: 'instance', id: instance.id },
+    })
+    return json(newMember, { status: 201 })
+  },
```

Comment:
I think instances can only be added to a group while stopped ([omicron source](https://github.com/oxidecomputer/omicron/blob/1a1f2c6865b7a31ef223e0bd3520ca628416dd35/nexus/db-queries/src/db/datastore/affinity.rs#L823-L876)). We should enforce that here.

In the UI I'm not so sure. It's a little weird to not see instances in the picker list that you expect to be there, just because they're running. Maybe we have to add copy to the add instance modal explaining that only stopped instances will show up in the list.

Reply:
We could show a message and disable the submit if it's running. The slight complicating factor is that at some point we might allow adding running instances, should the policy allow it – so we don't want to incept too strongly the current limitation.